tests/e2e/network/service_annotations.go

```diff
@@ -1097,7 +1115,32 @@ func validateLoadBalancerBackendPools(tc *utils.AzureTestClient, vmssName string
 
 			matches := backendIPConfigurationRE.FindStringSubmatch(*ipConfig.ID)
 			Expect(len(matches)).To(Equal(2))
-			Expect(matches[1]).To(Equal(vmssName))
+			backendPoolVMSSNames.Insert(matches[1])
 		}
 	}
+
+	return backendPoolVMSSNames
+}
+
+func validateLoadBalancerBackendPools(tc *utils.AzureTestClient, vmssName string, cs clientset.Interface, serviceName string, labels map[string]string, ns string, ports []v1.ServicePort, resourceGroupName string) {
```

Comment:
I think you can move `validateLoadBalancerBackendPools()` above `getVMSSNamesInLoadBalancerBackendPools` then we will have less code change.